### DISCUSSION TITLE
Is there a way to set a default value if the secret is not present in the Key Vault?

### DISCUSSION THREAD

**Discussion by ArronaxKP** [Azure/secrets-store-csi-driver-provider-azure#435] [Q&A] [answered]
If this is possible then can someone let me know how?

We have a scenario where we have new secrets being created and would like to make the "optional" and not cause the pod to fail on creation. 

From what I have read, the secret must exist in the Key Vault and if they are not present the container will fail to start.

**Comment by aramase** [comment, marked as answer]
It's not optional. All the secrets defined in the `SecretProviderClass` need to exist in the keyvault. If the mount fails, kubelet will retry the volume mount until it succeeds.
